### PR TITLE
Revert "Remove locking from DBs and rely on SQLite's locking (#211)"

### PR DESCRIPTION
**Pull request by daniellowell**
This reverts commit 7d0881309a7ef9bd072fceaa2ee854463be03566.

Suspected commit in Jenkins Failures.

### CODE REVIEW

**Comment by atamazov** [comment]
I've tried this on `wrw-igemm-v4r4xdlops-fp32-fix` branch, commit ad859f55c35140192c909bd683bd8e247b50981f (run #7 on Jenkins). Didn't help ((

File: src/sqlite_db.cpp [hunk 1]
```diff
@@ -277,34 +289,24 @@ SQLitePerfDb::SQLitePerfDb(const std::string& filename_,
...
         {
+            const auto lock = shared_lock(lock_file, GetLockTimeout());
+            MIOPEN_VALIDATE_LOCK(lock);
             // clang-format off
             const auto check_tables =
                 "SELECT name FROM sqlite_master "
                 "WHERE "
                   "type = 'table' AND "
-                  "(name = 'config');";
+                  "(name = 'config' OR name = 'perf_db');";
```
Comment (JehandadKhan) [inline src/sqlite_db.cpp:299]: This masks a bug, if only one of the tables is created.

Comment (daniellowell) [reply src/sqlite_db.cpp:299]: If this fixes the Jenkins issue then you can implement this fix, but leave off the rest of the locking changes.

**Comment by JehandadKhan** [comment]
@daniellowell  Did we find the root cause ?

**Pull request closed without merging**